kubetest/gke.go

```diff
@@ -0,0 +1,354 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package main / gke.go provides the Google Container Engine (GKE)
+// kubetest deployer via newGKE().
+//
+// TODO(zmerlynn): Pull this out to a separate package?
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+const (
+	defaultPool = "default"
+	e2eAllow    = "tcp:22,tcp:80,tcp:8080,tcp:30000-32767,udp:30000-32767"
+)
+
+var (
+	gkeCluster         = flag.String("gke-cluster", "", "(gke only) Cluster name. Must be set for GKE.")
```

Comment:
What do you think about making this either --gcp-cluster and/or --cluster? I'm planning to migrate CLUSTER_NAME to this in the near future